naming-conventions.md

```diff
@@ -168,18 +168,62 @@ consensus about this.
 
 ### Database Access
 
-We use the prefix `fetch` to signal retrieving data from the database.
+Database access functions should follow standard prefixes and suffixes for the
+following cases:
 
-Don't do this:
+- `fetch{...}` for a normal `SELECT`
+- `fetch{...}Exists(By{...})` for a `SELECT EXISTS`
+- `count{...}` for a `COUNT`
+
+And should follow a pattern that indicates:
+
+- What it returns, specifically if it returns a whole resource or just one field
+- What its arguments are, specifically if they are whole resources or fields
+- A correct pluralization of arguments and returned values
 
 ```
-getAnswers :: SqlReadT [Answer]
+{fetch|count}{Resource(Field)}(Exists)(By{Resource(Field)}(And{Resource(Field)}...))
 ```
 
-Do this:
+This is to avoid duplication by aiding code search through "obvious" names: if
+you know you need a `Foo` by its `barBat`, you should look for
+`fetchFooByBarBat` before writing it.
 
-```
-fetchAnswers :: SqlReadT [Answer]
+```hs
+-- Good
+fetchTeachersBySchoolId :: SchoolId -> SqlPersistT m [Entity Teacher]
+
+fetchTeacherIdsBySchool :: School
+
+-- NOTE: We don't go as far as distinguishing Entity or not in this naming, so
+-- these are also Good, with Resource ~ Teacher and Resource ~ Entity School
```

Comment:
I think if this were to lead to a conflict it can be resolved by:

- Upgrade return values of `e` to `Entity e` and discard the `entityKey` in callers
- Downgrade arguments from `Entity e` to `e` and discard the `entityKey` in callers

That said, I haven't seen this happen much.